src/serlo-editor/core/sub-document/editor.tsx

```diff
@@ -48,6 +49,31 @@ export function SubDocumentEditor({ id, pluginProps }: SubDocumentProps) {
     [focused, id, dispatch, document]
   )
 
+  // additional focus check to set focus when using tab navigation
+  const handleFocus = useCallback(
+    (e: React.FocusEvent) => {
+      // if after a short delay dom focus is not set inside focused plugin
+      // we overwrite it here (because it's probably because of tab navigation)
+      if (!document) return
+      if (['rows', 'exercise'].includes(document?.plugin)) return
+
+      const tabFocusTimeout = setTimeout(() => {
+        // fixes a bug in table plugin with disappearing buttons
+        if (e.target.nodeName?.toLowerCase() === 'button') return
+
+        // find closest document
+        const target = (e.target as HTMLDivElement).closest('[data-document]')
+
+        if (!focused && target === containerRef.current) {
+          dispatch(focus(id))
+        }
+      }, 10)
+
+      return () => clearTimeout(tabFocusTimeout)
```

Comment:
I thought this was `useEffect`-specific syntax 😄 Does it work for event handlers in general, or just for `useCallback`?

Reply:
Good catch, write it more out of habit. I don't think it works here.
I'm always a bit unsure how bad it would be to not clear a timeout like this.

Best and most complex:
Whenever a new focus event is triggered all old timeouts (on other plugins as well) get cleared (because we only want to focus the most current anyway)

In between:
I store the timer in a ref and clear it when unmounting the component.

Lazy:
We don't handle it at all.

Any votes? :)

Reply:
I rarely do `clearTimeout`s, to be honest. They are not recurring like `setEventHandler` or `clearInterval`, they happen once and clear themselves.

The users probably won't be changing the focus twice in 10 milliseconds. Maybe the only problematic behavior would be if the users hold Tab key? But even that works fine in this branch preview. So I vote for lazy, in this case. 😁

Reply:
I'll oblige :D